cmd/ack-generate/command/common.go

```diff
@@ -64,67 +66,93 @@ func isDirWriteable(fp string) bool {
 // ensureSDKRepo ensures that we have a git clone'd copy of the aws-sdk-go
 // repository, which we use model JSON files from. Upon successful return of
 // this function, the sdkDir global variable will be set to the directory where
-// the aws-sdk-go is found
-func ensureSDKRepo(cacheDir string) error {
+// the aws-sdk-go is found. It will also optionally fetch all the remote tags
+// and checkout the given tag.
+func ensureSDKRepo(
+	cacheDir string,
+	// A boolean instructing ensureSDKRepo whether to fetch the remote tags from
+	// the upstream repository
+	fetchTags bool,
+) error {
 	var err error
 	srcPath := filepath.Join(cacheDir, "src")
 	if err = os.MkdirAll(srcPath, os.ModePerm); err != nil {
 		return err
 	}
-	// clone the aws-sdk-go repository locally so we can query for API
-	// information in the models/apis/ directories
-	sdkDir, err = cloneSDKRepo(srcPath)
-	return err
-}
 
-// cloneSDKRepo git clone's the aws-sdk-go source repo into the cache and
-// returns the filepath to the clone'd repo. If the aws-sdk-go repository
-// already exists in the cache, it will checkout the current sdk-go version
-// mentionned in 'go.mod' file.
-func cloneSDKRepo(srcPath string) (string, error) {
-	sdkVersion, err := getSDKVersion()
-	if err != nil {
-		return "", err
+	// Clone repository if it doen't exist
+	sdkDir = filepath.Join(srcPath, "aws-sdk-go")
+	if _, err := os.Stat(sdkDir); os.IsNotExist(err) {
+		err = util.CloneRepository(context.Background(), sdkDir, sdkRepoURL)
+		if err != nil {
+			return fmt.Errorf("canot clone repository: %v", err)
+		}
 	}
-	clonePath := filepath.Join(srcPath, "aws-sdk-go")
-	if optRefreshCache {
-		if _, err := os.Stat(filepath.Join(clonePath, ".git")); !os.IsNotExist(err) {
-			cmd := exec.Command("git", "-C", clonePath, "fetch", "--all", "--tags")
-			if err = cmd.Run(); err != nil {
-				return "", err
-			}
-			cmd = exec.Command("git", "-C", clonePath, "checkout", "tags/"+sdkVersion)
-			return clonePath, cmd.Run()
+
+	// Fetch all tags
+	if fetchTags {
+		err = util.FetchRepositoryTags(context.Background(), sdkDir)
+		if err != nil {
+			return fmt.Errorf("cannot fetch tags: %v", err)
 		}
 	}
-	if _, err := os.Stat(clonePath); os.IsNotExist(err) {
-		cmd := exec.Command("git", "clone", "-b", sdkVersion, sdkRepoURL, clonePath)
-		return clonePath, cmd.Run()
+
+	// get sdkVersion and correct it (adds missing 'v')
+	sdkVersion, err := getSDKVersion("")
+	if err != nil {
+		return err
+	}
+	sdkVersion = correctSemanticVersion(sdkVersion)
+
+	// Now checkout the local repository.
```

Comment:
Comment doesn't match? `LoadRepository` doesn't `git checkout` - just loads into memory?

Reply:
Yeah it was more for the rest of the function body (`LoadRepository` + `CheckoutRepository`). Moving the comment